auth/oidc.go

```diff
@@ -423,3 +456,50 @@ func SetURLQueryParam(strURL, name, value string) (string, error) {
 	uri.RawQuery = rawQuery.Encode()
 	return uri.String(), nil
 }
+
+func (client *OIDCClient) GetProvider() (provider *oidc.Provider) {
+	for {
+		client.mux.Lock()
+		if client.provider != nil {
+			provider = client.provider
+			client.mux.Unlock()
+			break
+		}
+		base.Infof(base.KeyAuth, "Getting provider config ready; waiting for discovery sync.")
+		client.mux.Unlock()
+		time.Sleep(100 * time.Millisecond)
+	}
+	return
```

Comment:
Why do we need to loop whilst locking/unlocking to check? Can we not use an RWMutex for synchronization? (i.e. many readers blocked waiting for one writer to unlock)